src/npcmove.cpp

```diff
@@ -474,9 +473,10 @@ void npc::assess_danger()
             if( guy.lock() ) {
                 is_too_close |= too_close( critter.pos(), guy.lock()->pos(), def_radius );
             }
-            return is_too_close;
-        };
-        std::any_of( ai_cache.friends.begin(), ai_cache.friends.end(), test_too_close );
+            if( is_too_close ) {
+                break;
+            }
```

Comment:
Minor, but we could move this check before the `if ( guy.lock() ) { ...` check to save a little on performance since I think this is a hot code path.

Reply:
This wont change anything since is_too_close is updated only after `locking` each `guy`.
If I understand your suggestion correctly.

Reply:
It could start out true though if `dist <= def_radius` is true.

Reply:
I saw it. I will amend.